### PR TITLE
GitOps improvements

### PR DESCRIPTION
**Pull request by lukasvecerka23**
- move from webhook to webserver with one route POST /
- supports local git repo
- better docker image building, each image is build separately so when one build fails rest of the images will continue the building process
- if any build fails gitops returns 500 with build logs so anyone can check the error

### CODE REVIEW

File: bspump/jupyter/jupyter.py [hunk 1]
```diff
@@ -504,7 +506,6 @@ def webserver(app):
 
 if deploy_secret:
     import zipfile
-    from bspump.http.web.server import *
```
Comment (timthelion) [inline bspump/jupyter/jupyter.py:507]: This conditional import was intentional. Why import the thing if you're not gonna use it? If ruff complains then use a comment like `#noqa` or whatever you need to do to make ruff shut up.

File: pre/main.ipynb [hunk 2]
```diff
@@ -82,6 +82,20 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bae448a1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@register_connection\n",
+    "def connection(app):\n",
+    "    return bspump.http.web.server.WebServerConnection(app, \"WebServerConnection\", config={\n",
```
Comment (timthelion) [inline pre/main.ipynb:94]: Why not use the default connection that is created by `jupyter.py`? https://github.com/bitswan-space/BitSwan/blob/24b4a492b80d93ca97bea0adaed14f5e604653a7/bspump/jupyter/jupyter.py#L498

File: pre/main.ipynb [hunk 3]
```diff
@@ -141,8 +136,20 @@
    "outputs": [],
    "source": [
     "@step\n",
-    "def parse_json(event):\n",
-    "    return json.loads(event)"
+    "def check_secret(event):\n",
```
Comment (timthelion) [inline pre/main.ipynb:139]: OK, but I'd prefer to have a new web source built into bspump which checks the secret and doesn't even emit an event if the secret is invalid. This seems to be a pattern that we are repeating.

File: pre/main.py [hunk 4]
```diff
@@ -5,30 +5,35 @@
 config = None
 __bitswan_dev = False
 
-notebook_path = os.environ.get("JUPYTER_NOTEBOOK", "pipelines/main.ipynb")
+notebook_path = os.environ.get("JUPYTER_NOTEBOOK", "main.ipynb")
```
Comment (timthelion) [inline pre/main.py:8]: Hm? I'm confused how this change relates to the rest of the PR.

File: pre/main.ipynb (hunk 2 above)
Comment (lukasvecerka23) [reply pre/main.ipynb:94]: I don't understand this comment since I used `@register_connection`?

File: pre/main.ipynb (hunk 3 above)
Comment (lukasvecerka23) [reply pre/main.ipynb:139]: OK, I'll look at it

File: pre/main.py (hunk 4 above)
Comment (lukasvecerka23) [reply pre/main.py:8]: I had problems with deployment with this path...and it also affects deployments of our clients...but yeah it should've been part of different PR

**Pull request closed without merging**